### ISSUE TITLE
bazel_to_go.py (and linters.sh) causes further bazel-build failures

### ISSUE THREAD

**Issue by jmuk** [istio/mixer#1219]
After running ./bin/bazel_to_go.py, bazel build fails with the following error:
```
% bazel build //...
WARNING: /usr/local/google/home/mukai/.cache/bazel/_bazel_mukai/23d88ec2f9449b2c9d846b080b7cc444/external/bazel_tools/tools/build_defs/pkg/pkg.bzl:196:9: pkg_tar: renaming non-dict `files` attribute to `srcs`
WARNING: /usr/local/google/home/mukai/.cache/bazel/_bazel_mukai/23d88ec2f9449b2c9d846b080b7cc444/external/bazel_tools/tools/build_defs/pkg/pkg.bzl:196:9: pkg_tar: renaming non-dict `files` attribute to `srcs`
WARNING: /usr/local/google/home/mukai/.cache/bazel/_bazel_mukai/23d88ec2f9449b2c9d846b080b7cc444/external/bazel_tools/tools/build_defs/pkg/pkg.bzl:196:9: pkg_tar: renaming non-dict `files` attribute to `srcs`
ERROR: infinite symlink expansion detected
[start of symlink chain]
/usr/local/google/home/mukai/.cache/bazel/_bazel_mukai/23d88ec2f9449b2c9d846b080b7cc444/external/com_github_istio_mixer
/usr/local/google/home/mukai/gopath/src/istio.io/mixer
/usr/local/google/home/mukai/gopath/src/istio.io/mixer/vendor/github.com/istio/mixer
[end of symlink chain]
WARNING: Failed to get information about path, for vendor/github.com/istio/mixer, skipping: Infinite symlink expansion
ERROR: Infinite symlink expansion
INFO: Elapsed time: 0.234s
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Removing `vendor/` directory fixes this.

bazel_to_go.py is used by linters.sh, so running linters locally also causes this situation.

**Comment by douglas-reid** [comment]
@guptasu didn't we fix this in a recent PR, or is this new?

**Comment by guptasu** [comment]
This is new... Mandar and I did poke around this a bit last week. Not sure why this is happening.. I tried changes from 2 weeks back , was getting the same outcome.

**Comment by guptasu** [comment]
I had mentoned this on slack last week.. no one cared then

**Comment by douglas-reid** [comment]
I can't replicate this from HEAD.  Can we just add a filter on the `bazel_to_go.py` script?

**Comment by guptasu** [comment]
@douglas-reid what is a filter ?

So you ran
```
bazel build ...
./bin/bazel_to_go.py
bazel build ...
```

and everything worked ? Strange..

**Comment by douglas-reid** [comment]
yes. fwiw, i actually ran:

```
bazel clean
bazel build //cmd/...
bin/bazel_to_go.py
bazel build //cmd/...
```

by filter, i mean: update `bazel_to_go.py` not to symlink anything related to `com_github_istio_mixer`

**Comment by douglas-reid** [comment]
Does https://github.com/istio/mixer/pull/1226 fix your issues?

**Comment by jmuk** [comment]
It's okay with `//cmd/...`, but it reproduces if you do with `bazel build //...`. Don't know why.
 #1226 looks like solving this situation.

**Comment by jmuk** [comment]
fixed by #1226
